### ISSUE TITLE
Should we shuffle the dataset every iteration

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#36]
## Introduction
Check if there is any gains in test accuracy by shuffling data every iteration.
Seems that the best is to shuffle the data only at each epoch, not every iteration.

## References
* [Neural Networks: Tricks of the Trade](https://github.com/leonardoaraujosantos/DLMatFramework/files/834228/TircksTrade.pdf)
* [Practical Recommendations for Gradient-Based Training of Deep Models](https://github.com/leonardoaraujosantos/DLMatFramework/files/834232/1206.5533.pdf)
* http://stackoverflow.com/questions/40816721/should-i-shuffle-the-data-to-train-a-neural-network-using-backpropagation

**Comment by Burton2000** [comment]
Commited changes to the way batch is shuffled. Rather than shuffle every time we grab a batch we shuffle at the start of each epoch. 

Tested with mnist and obtains 99% validation accuracy so has not made things worse.
